src/composables/transactionList.ts

```diff
@@ -0,0 +1,329 @@
+      return [];
+    }
+    const { pending, loaded } = getAccountTransactionsState(address);
+
+    return [...loaded, ...(pending[nodeNetworkId.value!] || [])];
+  };
+
+  function setPendingTransactionSentByHash(address: Encoded.AccountAddress, hash: string) {
+    const index = transactions.value[address].pending[nodeNetworkId.value!]
+      ?.findIndex((transaction) => transaction.hash === hash);
+
+    if (index !== -1) {
+      transactions.value[address]
+        .pending[nodeNetworkId.value!][index][TX_DIRECTION.sent] = true;
+    }
+  }
+
+  function removePendingTransactionByAccount(address: Encoded.AccountAddress, hash: string) {
+    const pendingTransactionForAccount: ITransaction[] = (
+      transactions.value[address]?.pending?.[nodeNetworkId.value!] || []
+    );
+
+    if (pendingTransactionForAccount.length) {
+      transactions.value[address].pending[nodeNetworkId.value!] = (
+        pendingTransactionForAccount.filter((transaction) => transaction.hash !== hash)
+      );
+    }
+  }
+
+  async function waitTransactionMined(address: Encoded.AccountAddress, hash?: string) {
+    if (hash) {
+      try {
+        const aeSdk = await getAeSdk();
+        await aeSdk.poll(hash as Encoded.TxHash);
+        setPendingTransactionSentByHash(address, hash);
+      } catch (e) {
+        removePendingTransactionByAccount(address, hash);
+      }
+    }
+  }
+
+  function upsertCustomPendingTransactionForAccount(
+    address: Encoded.AccountAddress,
+    transaction: ITransaction,
+  ) {
+    ensureAccountTransactionStateExists(address);
+    transactions.value[address].pending[nodeNetworkId.value!] = [transaction];
+    waitTransactionMined(address, transaction.hash);
+  }
+
+  async function fetchTransactionsFromMiddleware(
+    transactionState: ITransactionsState,
+    address: Encoded.AccountAddress,
+    recent: boolean,
+    limit: number,
+  ) {
+    const url = (recent || transactionState.nextPageUrl === '')
+      ? `/v2/accounts/${address}/activities?limit=${limit}`
+      : transactionState.nextPageUrl!;
+
+    try {
+      const { data, next } = await fetchFromMiddlewareCamelCased(url);
+      if (!recent) {
+        setTransactionsNextPage(address, next);
+      }
+      return data;
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchPendingTransactions(address: Encoded.AccountAddress) {
+    const sdk = await getAeSdk();
+
+    try {
+      const fetchedPendingTransaction = (
+        await sdk.api.getPendingAccountTransactionsByPubkey(address)
+      );
+      const transactionState = getAccountTransactionsState(address);
+
+      return JsonBig.parse(JsonBig.stringify(
+        fetchedPendingTransaction?.transactions || [],
+      ))
+        .filter((transaction: ITransaction) => !transactionState.pending[nodeNetworkId.value!]
+          ?.find((tx) => tx?.hash === transaction?.hash))
+        .map((transaction: ITransaction) => ({ ...transaction, pending: true }));
```

Comment:
At first glance it looks like it has wrong indentation here - consider refactoring
```suggestion
        .filter(
          (transaction: ITransaction) => !transactionState.pending[nodeNetworkId.value!]?.find(
            (tx) => tx?.hash === transaction?.hash
          )
        )
        .map((transaction: ITransaction) => ({ ...transaction, pending: true }));
```